certified-connectors/TractionGuest/apiDefinition.swagger.json

```diff
@@ -0,0 +1,582 @@
+      "title": "Root Type for Watchlist",
+      "description": "The root of the Watchlist type's schema.",
+      "type": "object",
+      "properties": {
+        "internal": {
+          "type": "boolean",
+          "x-ms-summary": "Internal watchlist?"
+        },
+        "color": {
+          "type": "string",
+          "x-ms-summary": "Watchlist color"
+        },
+        "text": {
+          "type": "string",
+          "x-ms-summary": "Watchlist description (external only)"
+        },
+        "source_information_url": {
+          "type": "string",
+          "x-ms-summary": "Watchlist information (external only)"
+        },
+        "source": {
+          "type": "string",
+          "x-ms-summary": "Watchlist source"
+        }
+      }
+    },
+    "WebhookSignin": {
+      "required": ["acknowledged", "signin_timestamp", "watchlists", "hosts"],
+      "type": "object",
+      "properties": {
+        "id": {
+          "format": "int32",
+          "type": "integer",
+          "x-ms-summary": "Signin ID"
+        },
+        "acknowledged": {
+          "type": "boolean",
+          "x-ms-summary": "Guest acknowledged?"
+        },
+        "guest_first_name": {
+          "type": "string",
+          "x-ms-summary": "Guest first name"
+        },
+        "guest_last_name": {
+          "type": "string",
+          "x-ms-summary": "Guest last name"
+        },
+        "guest_email": {
+          "type": "string",
+          "x-ms-summary": "Guest email"
+        },
+        "company": {
+          "type": "string",
+          "x-ms-summary": "Guest company"
+        },
+        "visit_reason": {
+          "type": "string",
+          "x-ms-summary": "Reason for visit"
+        },
+        "photo_thumb_url": {
+          "type": "string",
+          "x-ms-summary": "Link to photo thumbnail of guest"
+        },
+        "photo_medium_url": {
+          "type": "string",
+          "x-ms-summary": "Link to photo of guest"
+        },
+        "location_name": {
+          "type": "string",
+          "x-ms-summary": "Location name"
+        },
+        "signin_timestamp": {
+          "format": "date-time",
+          "type": "string",
+          "x-ms-summary": "Signin time"
+        },
+        "report_url": {
+          "type": "string",
+          "x-ms-summary": "Link to alert security"
+        },
+        "badge_number": {
+          "type": "string",
+          "x-ms-summary": "Guest badge ID"
+        },
+        "watchlists": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Watchlist"
+          },
+          "x-ms-summary": "Guest's watchlist colors"
+        },
+        "hosts": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Host"
+          }
+        },
+        "photo_filename": {
+          "type": "string",
+          "x-ms-summary": "Photo filename, if available"
+        },
+        "photo": {
+          "format": "binary",
```

Comment:
Since "photo" is a property within JSON object, it is unlikely to pass arbitrary binary content here correctly.
If the content is passed as base64-encoded string, it should have "format":"byte". If it s a simple string with some other type of encoding for octet-stream - "format" should be removed.

Reply:
As discussed, we will have a rule in the swagger validator.